### PR TITLE
Add Kotlin "Spring"

### PR DESCRIPTION
**Pull request by mmaryo**

### CODE REVIEW

File: kotlin/spring/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+plugins {
+	id("org.springframework.boot") version "3.3.1"
+	id("io.spring.dependency-management") version "1.1.5"
+	kotlin("jvm") version "1.9.24"
+	kotlin("plugin.spring") version "1.9.24"
```
Comment (waghanza) [inline kotlin/spring/build.gradle.kts:5]:
```suggestion
	id("org.springframework.boot") version "3.3.+"
	id("io.spring.dependency-management") version "+"
	kotlin("jvm") version "1.9.+"
	kotlin("plugin.spring") version "1.9.+"
```

to make sure we use the patched version of each

File: kotlin/spring/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+dependencies {
+	implementation("org.springframework.boot:spring-boot-starter-web")
+	implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
+	implementation("org.jetbrains.kotlin:kotlin-reflect")
+}
+
+kotlin {
+	compilerOptions {
+		freeCompilerArgs.addAll("-Xjsr305=strict")
```
Comment (waghanza) [inline kotlin/spring/build.gradle.kts:29]: is it something specific to kotlin / spring ? I mean this seems to be a jvm option, and then makes the results unfaire between jvm based frameworks

File: kotlin/spring/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	}
+}
+
+tasks.withType<Test> {
```
Comment (waghanza) [inline kotlin/spring/build.gradle.kts:33]: required ?

File: kotlin/spring/gradle/wrapper/gradle-wrapper.jar
Comment (waghanza) [inline kotlin/spring/gradle/wrapper/gradle-wrapper.jar:1]: could this file be in .gitignore, no need to version it, imho

File: kotlin/spring/gradle/wrapper/gradle-wrapper.properties
Comment (waghanza) [inline kotlin/spring/gradle/wrapper/gradle-wrapper.properties:1]: could this file be in .gitignore, no need to version it, imho

**Review by waghanza — CHANGES_REQUESTED** [review]

File: kotlin/spring/build.gradle.kts (hunk 2 above)
Comment (mmaryo) [reply kotlin/spring/build.gradle.kts:29]: it's about null safty and by default with spring starter, but not useful in this case https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/boot-features-kotlin.html#boot-features-kotlin-null-safety

File: kotlin/spring/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+server.port=3000
+server.error.whitelabel.enabled=false
```
Comment (waghanza) [inline kotlin/spring/src/main/resources/application.properties:2]: What is this for ?

**Review by waghanza — COMMENTED** [review]
Quick question

File: kotlin/spring/src/main/resources/application.properties (hunk 4 above)
Comment (waghanza) [reply kotlin/spring/src/main/resources/application.properties:2]: https://www.baeldung.com/spring-boot-custom-error-page

Maybe the accurate to the same for spring/spring webflux in java

**Review by waghanza — COMMENTED** [review]
Maybe replicate the same behaviour

**Review by waghanza — APPROVED** [review]

File: kotlin/spring/src/main/resources/application.properties (hunk 4 above)
Comment (mmaryo) [reply kotlin/spring/src/main/resources/application.properties:2]: same as spring java

**Pull request merged**